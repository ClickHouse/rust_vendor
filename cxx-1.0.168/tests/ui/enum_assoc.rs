#[cxx::bridge]
mod ffi {
    enum Enum {
        Variant,
    }
    extern "Rust" {
        #[Self = "Enum"]
        fn f();
    }
}

impl ffi::Enum {
    fn f() {}
}

fn main() {}
