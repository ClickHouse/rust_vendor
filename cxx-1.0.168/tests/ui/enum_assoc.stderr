error: unsupported self type; C++ does not allow member functions on enums
 --> tests/ui/enum_assoc.rs:7:18
  |
7 |         #[Self = "Enum"]
  |                  ^^^^^^

error[E0433]: failed to resolve: use of unresolved module or unlinked crate `ffi`
  --> tests/ui/enum_assoc.rs:12:6
   |
12 | impl ffi::Enum {
   |      ^^^ use of unresolved module or unlinked crate `ffi`
   |
   = help: if you wanted to use a crate named `ffi`, use `cargo add ffi` to add it to your `Cargo.toml`
help: consider importing this module
   |
 1 + use cxx_test_suite::ffi;
   |
